flake.nix

```diff
@@ -0,0 +1,187 @@
+{
+  description = "dhcpv6macd";
+
+  # Nixpkgs / NixOS version to use.
+  inputs.nixpkgs.url = "nixpkgs/nixos-unstable-small";
+  inputs.nix-filter.url = "github:numtide/nix-filter";
+
+  outputs = { self, nixpkgs, nix-filter }:
+    let
+
+      # to work with older version of flakes
+      lastModifiedDate = self.lastModifiedDate or self.lastModified or "19700101";
+
+      # Generate a user-friendly version number.
+      version = builtins.substring 0 8 lastModifiedDate;
+
+      # System types to support.
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" "aarch64-linux" "aarch64-darwin" ];
+
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = nixpkgs.lib.genAttrs supportedSystems;
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
+
+    in
+    {
+      devShell = forAllSystems
+        (system:
+          let
+            pkgs = nixpkgsFor.${system};
+          in
+          pkgs.mkShell {
+            buildInputs = with pkgs; [
+              codespell
+              eclint
+              go
+              nixpkgs-fmt
+            ];
+          });
+
+      # Provide some binary packages for selected system types.
+      packages = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+        in
+        {
+          default = pkgs.buildGoModule {
+            pname = "dhcpv6macd";
+            inherit version;
+            # In 'nix develop', we don't need a copy of the source tree
```

Comment:
How is `nix develop` relevant here? Are we expecting people to `nix develop .#packages.x86_64-linux.packages.default`?